js/extension/plugins/LongitudinalProfileModule.jsx

```diff
@@ -0,0 +1,137 @@
+ * LICENSE file in the root directory of this source tree.
+*/
+
+import {connect} from "react-redux";
+import { createSelector } from 'reselect';
+
+import { name } from '../../../config';
+
+import Main from "@js/extension/components/Main";
+import Menu from "@js/extension/components/Menu";
+import GlobalSpinner from "@mapstore/components/misc/spinners/GlobalSpinner/GlobalSpinner";
+
+import * as epics from '@js/extension/epics';
+
+import {
+    dataSourceMode, distanceSelector,
+    infosSelector,
+    isActiveMenu,
+    isDockOpen, isInitialized,
+    isParametersOpen, isSupportedLayer,
+    pointsSelector, referentialSelector,
+    projectionSelector,
+    isMaximized, isLoading
+} from "@js/extension/selectors";
+import longitudinal from '@js/extension/reducers/longitudinal';
+
+import {
+    closeDock,
+    setup,
+    tearDown,
+    toggleMode,
+    changeReferential,
+    changeDistance,
+    changeGeometry,
+    toggleMaximize,
+    addMarker,
+    hideMarker
+} from "@js/extension/actions/longitudinal";
+import { setControlProperty } from "@mapstore/actions/controls";
+import { warning } from '@mapstore/actions/notifications';
+
+import {exportCSV} from "@mapstore/actions/widgets";
+import {boundingSidebarRectSelector, mapLayoutValuesSelector} from "@mapstore/selectors/maplayout";
+import {currentLocaleSelector, currentMessagesSelector} from "@mapstore/selectors/locale";
+
+import {getSelectedLayer} from "@mapstore/selectors/layers";
+import '../assets/style.css';
+
+const selector = (state) => ({
+    initialized: isInitialized(state),
+    size: 550,
+    loading: isLoading(state),
+    dataSourceMode: dataSourceMode(state),
+    isParametersOpen: isParametersOpen(state),
+    showDock: isDockOpen(state),
+    infos: infosSelector(state),
+    points: pointsSelector(state),
+    projection: projectionSelector(state),
+    referential: referentialSelector(state),
+    distance: distanceSelector(state),
+    dockStyle: mapLayoutValuesSelector(state, { height: true, right: true }, true),
+    helpStyle: mapLayoutValuesSelector(state, { right: true }, false),
+    selectedLayer: getSelectedLayer(state),
+    isSupportedLayer: isSupportedLayer(state),
+    messages: currentMessagesSelector(state),
+    currentLocale: currentLocaleSelector(state),
+    maximized: isMaximized(state),
+    boundingRect: boundingSidebarRectSelector(state)
+});
+
+export const LongitudinalNav = connect((state) => ({
+    nav: false,
+    className: "square-button",
+    menuIsActive: isActiveMenu(state),
+    isActive: isActiveMenu(state),
+    dataSourceMode: dataSourceMode(state),
+    isParametersOpen: isParametersOpen(state),
+    initialized: isInitialized(state)
+}), {
+    onActivateTool: setControlProperty.bind(this, "longitudinalProfile", "enabled", true),
+    onToggleSourceMode: toggleMode,
+    onToggleParameters: setControlProperty.bind(this, "longitudinalProfileParameters", "enabled", true, true)
+})(Menu);
+
+export default {
```

Comment:
Could you check if it's working?
Shouldn't this be exported with createPlugin?
see https://github.com/geosolutions-it/MapStoreExtension/blob/master/js/extension/plugins/Module.jsx#L11